src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -472,6 +575,43 @@ private ICredentialProvider GetCredentialProvider(CommandSettings command, strin
             return provider;
         }
 
+        private void AssertAdminAccess(bool unConfigure = false)
```

Comment:
move this check to respective *configManager(s).

---
In reply to: [117653784](https://github.com/Microsoft/vsts-agent/pull/978#discussion_r117653784) [](ancestors = 117653784)